src/app/components/chat/window/output/AppChatWindowOutputItemDetails.vue

```diff
@@ -0,0 +1,91 @@
+<script lang="ts" setup>
+import { computed, PropType, toRefs } from 'vue';
+import { formatDate } from '../../../../../_common/filters/date';
+import AppJolticon from '../../../../../_common/jolticon/AppJolticon.vue';
+import { vAppTooltip } from '../../../../../_common/tooltip/tooltip-directive';
+import { useChatStore } from '../../chat-store';
+import { retryFailedQueuedMessage } from '../../client';
+import { ChatMessage } from '../../message';
+import { ChatRoom } from '../../room';
+
+const props = defineProps({
+	message: {
+		type: Object as PropType<ChatMessage>,
+		required: true,
+	},
+	room: {
+		type: Object as PropType<ChatRoom>,
+		required: true,
+	},
+	timestampMarginLeft: {
+		type: Number,
+		default: 0,
+	},
+	showAmPm: {
+		type: Boolean,
+	},
+});
+
+const { message, room, showAmPm } = toRefs(props);
+
+const chatStore = useChatStore();
+
+const loggedOn = computed(() => {
+	if (!room.value.shouldShowTimestamp) {
+		return null;
+	}
+
+	const result = {
+		template: formatDate(message.value.logged_on, 'shortTime'),
+		tooltip: formatDate(message.value.logged_on, 'medium'),
+	};
+
+	if (!showAmPm.value) {
+		result.template = result.template.replace(/( )?(am|pm)/i, '');
```

Comment:
not good for localization. instead add some new ShortcutFormats in formatDate for shortTimeNoMeridiem and mediumNoMeridiem